### PR TITLE
NMS-15302: Added security-related HTTP headers

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-15302

### CODE REVIEW

**Review by RangerRick — CHANGES_REQUESTED** [review]
This should have a doc component at least in the release notes pointing out that some content restrictions are tighter now, since it's possible people are using dashboards or embedding external things in stuff defined in `org.opennms.web.console.centerUrl` and such.

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2023 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 2023 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java:11]:
```suggestion
 * by the Free Software Foundation, either version 3 of the license,
```

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java:16]:
```suggestion
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
```

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java:20]:
```suggestion
 * along with OpenNMS(R). If you did not, see:
```

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java:24]: does this email still work? Should we update it to opennms.com?

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ *     http://www.opennms.org/
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java:25]: can we remove the .org URL?

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddStrictTransportSecurityHeaderFilter.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2023 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 2023 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddStrictTransportSecurityHeaderFilter.java:11]:
```suggestion
 * by the Free Software Foundation, either version 3 of the license,
```

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddStrictTransportSecurityHeaderFilter.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddStrictTransportSecurityHeaderFilter.java:16]:
```suggestion
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
```

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddStrictTransportSecurityHeaderFilter.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddStrictTransportSecurityHeaderFilter.java:24]: see earlier comment about .com vs. .org

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddStrictTransportSecurityHeaderFilter.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+ *     http://www.opennms.org/
```
Comment (Bonrob2) [inline opennms-web-api/src/main/java/org/opennms/web/servlet/AddStrictTransportSecurityHeaderFilter.java:25]: can we remove .org URL?

**Review by Bonrob2 — COMMENTED** [review]
I think last time I edited the copyright text there were issues because of the build process, so feel free to ignore my comments. The .org should be updated at some point, perhaps under a different ticket?

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java (hunk 4 above)
Comment (christianpape) [reply opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java:24]: I hope so. This email is present in each and every header.

File: opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java (hunk 5 above)
Comment (christianpape) [reply opennms-web-api/src/main/java/org/opennms/web/servlet/AddHeaderFilter.java:25]: I don't think so.

**Review by Bonrob2 — APPROVED** [review]
text LGTM! Thank you!

**Review by Bonrob2 — APPROVED** [review]
committed one minor change that was missed. Text LGTM!

**Review by aramosvizcarra — DISMISSED** [review]
LGTM

**Review by Bonrob2 — APPROVED** [review]
LGTM!

**Review by RangerRick — APPROVED** [review]

**Pull request merged**